adapters/sharethrough/sharethrough.go

```diff
@@ -1,62 +1,103 @@
 	"fmt"
-	"net/http"
-	"regexp"
-
 	"github.com/mxmCherry/openrtb/v15/openrtb2"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/config"
 	"github.com/prebid/prebid-server/errortypes"
 	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/version"
+	"net/http"
+	"strings"
 )
 
-const supplyId = "FGMrCMMc"
-const strVersion = 8
+var strVersion = "10.0"
+
+type SharethroughAdapter struct {
+	endpoint string
+}
 
-// Builder builds a new instance of the Sharethrough adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
 	bidder := &SharethroughAdapter{
-		AdServer: StrOpenRTBTranslator{
-			UriHelper: StrUriHelper{BaseURI: config.Endpoint, Clock: Clock{}},
-			Util:      Util{Clock: Clock{}},
-			UserAgentParsers: UserAgentParsers{
-				ChromeVersion:    regexp.MustCompile(`Chrome\/(?P<ChromeVersion>\d+)`),
-				ChromeiOSVersion: regexp.MustCompile(`CriOS\/(?P<chromeiOSVersion>\d+)`),
-				SafariVersion:    regexp.MustCompile(`Version\/(?P<safariVersion>\d+)`),
-			},
-		},
+		endpoint: config.Endpoint,
 	}
 	return bidder, nil
 }
 
-type SharethroughAdapter struct {
-	AdServer StrOpenRTBInterface
-}
-
 func (a SharethroughAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var reqs []*adapters.RequestData
+	var requests []*adapters.RequestData
+	var errors []error
 
-	if request.Site == nil {
-		return nil, []error{fmt.Errorf("request must include a site; in-app placements are not supported")}
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	modifiableSource := openrtb2.Source{}
+	if request.Source != nil {
+		modifiableSource = *request.Source
+	}
+	var sourceExt map[string]interface{}
+	if err := json.Unmarshal(modifiableSource.Ext, &sourceExt); err == nil {
+		sourceExt["str"] = strVersion
+		sourceExt["version"] = version.Ver
+	} else {
+		sourceExt = map[string]interface{}{"str": strVersion, "version": version.Ver}
 	}
-	var domain = Util{}.parseDomain(request.Site.Page)
+	modifiableSource.Ext, _ = json.Marshal(sourceExt)
+	request.Source = &modifiableSource
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		// Extract Sharethrough Params
+		var strImpExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &strImpExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		var strImpParams openrtb_ext.ExtImpSharethrough
+		if err := json.Unmarshal(strImpExt.Bidder, &strImpParams); err != nil {
+			errors = append(errors, err)
+			continue
+		}
 
-	for i := 0; i < len(request.Imp); i++ {
-		req, err := a.AdServer.requestFromOpenRTB(request.Imp[i], request, domain)
+		// Convert Floor into USD
+		if imp.BidFloor > 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" {
```

Comment:
It's more efficient to use the case insensitive `EqualFold` instead of converting the bid floor currency to an uppercase string.

Reply:
I copy pasted from the example here: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#currency

You might wanna update the example in the doc, or future adapters might have the same inefficiency ;)

Reply:
Fair point. I'll update the example tomorrow.

Reply:
I'm assuming I don't even need `imp.BidFloorCur != ""` in the test either, the `EqualFold` should handle it juste fine, so I'm removing it, let me know if that's a mistake